filebeat/input/filestream/internal/input-logfile/clean.go

```diff
@@ -33,23 +35,24 @@ type cleaner struct {
 
 // run starts a loop that tries to clean entries from the registry.
 // The cleaner locks the store, such that no new states can be created
-// during the cleanup phase. Only resources that are finished and whos TTL
+// during the cleanup phase. Only resources that are finished and whose TTL
 // (clean_timeout setting) has expired will be removed.
 //
 // Resources are considered "Finished" if they do not have a current owner (active input), and
 // if they have no pending updates that still need to be written to the registry file after associated
 // events have been ACKed by the outputs.
+//
 // The event acquisition timestamp is used as reference to clean resources. If a resources was blocked
-// for a long time, and the life time has been exhausted, then the resource will be removed immediately
+// for a long time, and the lifetime has been exhausted, then the resource will be removed immediately
 // once the last event has been ACKed.
 func (c *cleaner) run(canceler unison.Canceler, store *store, interval time.Duration) {
 	started := time.Now()
 	err := timed.Periodic(canceler, interval, func() error {
 		gcStore(c.log, started, store)
 		return nil
 	})
-	if err != nil {
-		c.log.Errorf("failed to start the registry cleaning routine: %w", err)
+	if err != nil && !errors.Is(err, context.Canceled) {
+		c.log.Errorw("failed running periodic registry cleaning routine", "error", err)
```

Comment:
I believe it's better to keep the original format of the message which adds the error to the `message` field of the log entry. That's the patter used in most, if not all, Beats codebase.
```suggestion
		c.log.Errorf("failed to start the registry cleaning routine: %s", err)
```

Reply:
The format here creates log messages in the ECS error format. The logger is specifically setup to do this as per https://github.com/elastic/ecs-logging-go-zap/blob/main/docs/introduction.asciidoc. Why should Beats not follow the Elastic Common Schema for its own logging?

From experience ingesting logs from many products, I would prefer more structure to the data. Additionally using structured logging methods in zap is usually better because it avoids the allocations associated for the sprintf. And finally the structured logging methods are used in many places including [instances](https://github.com/search?q=repo%3Aelastic%2Fbeats+path%3A*.go+%2F%5C.%28Error%7CWarn%7CInfo%7CDebug%29w%5C%28.*%2C+%22error%22%2C+.*%2F+&type=code&p=1) that log errors.

Reply:
+1 to using the ECS compatible format where possible, Beats came before ECS which is part of why we don't do this already.

Reply:
@belimawr Could you please respond to this thread. I would like to move this discussion along because it is now blocking https://github.com/elastic/beats/pull/35798 which is important for aiding in SDH investigations.

Reply:
As @cmacknz said, Beats came before ECS and most of its codebase does not follow ECS (at least the ones maintained by the Elastic-Agent/Beats team), and I rather keep consistency within the codebase, specially within a package. On a quick look into your search, most of it is not maintained by the Beats team, and [searching only on filestream input](https://github.com/search?q=repo%3Aelastic%2Fbeats+path%3A*.go+%2F%5C.%28Error%7CWarn%7CInfo%7CDebug%29w%5C%28.*%2C+%22error%22%2C+.*%2F++path%3A%2F%5Efilebeat%5C%2Finput%5C%2Ffilestream%2F&type=code) I found no instances of ECS being used for error.

However I do see your point, @cmacknz's and @pierrehilbert, so it's not a blocker.